index.js

```diff
@@ -228,51 +200,57 @@ function rateLimitRequestHandler (params, pluginComponent) {
       }
     }
 
-    const timeLeft = Math.floor(ttl / 1000)
+    timeLeftInSeconds = Math.floor(ttl / 1000)
 
-    if (current <= maximum) {
-      if (params.addHeadersOnExceeding[params.labels.rateLimit]) { res.header(params.labels.rateLimit, maximum) }
-      if (params.addHeadersOnExceeding[params.labels.rateRemaining]) { res.header(params.labels.rateRemaining, maximum - current) }
-      if (params.addHeadersOnExceeding[params.labels.rateReset]) { res.header(params.labels.rateReset, timeLeft) }
+    if (current <= max) {
+      if (params.addHeadersOnExceeding[params.labels.rateLimit]) { res.header(params.labels.rateLimit, max) }
+      if (params.addHeadersOnExceeding[params.labels.rateRemaining]) { res.header(params.labels.rateRemaining, max - current) }
+      if (params.addHeadersOnExceeding[params.labels.rateReset]) { res.header(params.labels.rateReset, timeLeftInSeconds) }
 
-      params.onExceeding?.(req, key)
+      params.onExceeding(req, key)
 
       return
     }
 
-    params.onExceeded?.(req, key)
+    params.onExceeded(req, key)
 
-    if (params.addHeaders[params.labels.rateLimit]) { res.header(params.labels.rateLimit, maximum) }
+    if (params.addHeaders[params.labels.rateLimit]) { res.header(params.labels.rateLimit, max) }
     if (params.addHeaders[params.labels.rateRemaining]) { res.header(params.labels.rateRemaining, 0) }
-    if (params.addHeaders[params.labels.rateReset]) { res.header(params.labels.rateReset, timeLeft) }
+    if (params.addHeaders[params.labels.rateReset]) { res.header(params.labels.rateReset, timeLeftInSeconds) }
     if (params.addHeaders[params.labels.retryAfter]) {
-      const resetAfterTime = (params.enableDraftSpec ? timeLeft : params.timeWindowInSeconds)
+      const resetAfterTime = (params.enableDraftSpec ? timeLeftInSeconds : params.timeWindowInSeconds)
       res.header(params.labels.retryAfter, resetAfterTime)
     }
 
-    const code = params.ban && current - maximum > params.ban ? 403 : 429
+    const code = params.ban && current - max > params.ban ? 403 : 429
     const respCtx = {
       statusCode: code,
-      after,
-      max: maximum,
-      ttl
+      ban: false,
+      max,
+      ttl,
+      after: ms.format(params.timeWindow, true)
+
     }
 
     if (code === 403) {
       respCtx.ban = true
-      params.onBanReach?.(req, key)
+      params.onBanReach(req, key)
     }
 
     throw params.errorResponseBuilder(req, respCtx)
   }
 }
 
+function defaultKeyGenerator (req) { return req.ip }
```

Comment:
before it was an anonymous function having not a this scope (maybe little bit faster)

Reply:
I actually think about this more than necessary and am of the same opinion even though I haven't throughly tested it 😂

As you know normal functions have 2 extra objects, `this` and `arguments`, so in theory arrow functions don't have this baggage and if `this` is not of concern, it could mean better

But the engine probably optimizes it when it doesn't see the use of this inside a function